smartgymapi/handlers/cardio_activity.py

```diff
@@ -0,0 +1,80 @@
+import logging
+from datetime import datetime
+
+from marshmallow import ValidationError
+from pyramid.httpexceptions import HTTPBadRequest, HTTPInternalServerError
+from pyramid.view import view_defaults, view_config
+
+from smartgymapi.models import persist, commit, rollback, delete
+from smartgymapi.lib.factories.cardio_acitivty import CardioActivityFactory, is_cardio_activity_active
+from smartgymapi.lib.validation.cardio_activity import CardioActivitySchema
+from smartgymapi.models.cardio_activity import CardioActivity
+
+log = logging.getLogger(__name__)
+
+
+@view_defaults(containment=CardioActivityFactory,
+               permission='public',
+               renderer='json')
+class RESTCardioActivty(object):
+    def __init__(self, request):
+        self.request = request
+
+    @view_config(context=CardioActivityFactory, request_method='GET')
+    def list(self):
+        try:
+            activity_id = self.request.GET['activity_id']
+            return CardioActivitySchema(many=True).dump(self.request.context.list_cardio_activities(activity_id)).data
+        except KeyError as e:
+            raise HTTPBadRequest(json={'message': 'Query parameter %s is needed' % e})
+
+    @view_config(context=CardioActivity, request_method='GET')
+    def get(self):
+        return CardioActivity().dump(self.request.context).data
+
+    @view_config(context=CardioActivityFactory, request_method='POST')
+    def post(self):
+        self.save(CardioActivity())
+
+    @view_config(context=CardioActivity, request_method='PUT')
+    def put(self):
+        cardio_activity = self.request.context
+        if cardio_activity.end_date is None:
+            cardio_activity.end_date = datetime.now()
+        else:
+            log.critical('Cardio activity is already ended')
```

Comment:
Instead of logging this you should probably return this in the `HTTPBadRequest` exception so that the user knows what's up.